File: templates/Uwp/_composition/CaliburnMicro/Project.TabbedPivot/Behaviors/PivotBehavior.cs
using System;
using System.Linq;
using Microsoft.Xaml.Interactivity;
using Windows.UI.Xaml.Controls;
using Caliburn.Micro;
using wts.ItemName.Helpers;

namespace wts.ItemName.Behaviors
{
    public class PivotBehavior : Behavior<Pivot>
    {
        protected override void OnAttached()
        {
            base.OnAttached();
            AssociatedObject.SelectionChanged += OnSelectionChanged;
        }

        protected override void OnDetaching()
        {
            base.OnDetaching();
            AssociatedObject.SelectionChanged -= OnSelectionChanged;
        }

        private async void OnSelectionChanged(object sender, SelectionChangedEventArgs e)
        {
            var removedItem = e.RemovedItems.Cast<Screen>()
                .Select(i => GetPivotPage(i)).FirstOrDefault();

            var addedItem = e.AddedItems.Cast<Screen>()
                .Select(i => GetPivotPage(i)).FirstOrDefault();

            if (removedItem != null)
            {
                await removedItem.OnPivotUnselectedAsync();
            }

            if (addedItem != null)
            {
                await addedItem?.OnPivotSelectedAsync();
            }
        }

        private static IPivotPage GetPivotPage(Screen screen)
        {
            if (screen.GetView() is IPivotPage pivotPage)
            {
                return pivotPage;
            }

            return null;
        }
    }
}

File: templates/Uwp/_composition/MVVMLight/Page.TabbedPivot.AddBehaviorVB/Behaviors/PivotBehavior.vb
Imports System
Imports Param_ItemNamespace.Helpers
Imports Microsoft.Xaml.Interactivity

Namespace Behaviors
    Public Class PivotBehavior
        Inherits Behavior(Of Pivot)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.SelectionChanged, AddressOf OnSelectionChanged
        End Sub

        Protected Overrides Sub OnDetaching()
            MyBase.OnDetaching()
            RemoveHandler AssociatedObject.SelectionChanged, AddressOf OnSelectionChanged
        End Sub

        Private Async Sub OnSelectionChanged(sender As Object, e As SelectionChangedEventArgs)
            Dim removedItem = e.RemovedItems.Cast(Of PivotItem).Select(Function(pi) pi.GetPage(Of IPivotPage)).FirstOrDefault()
            Dim addedItem = e.AddedItems.Cast(Of PivotItem).Select(Function(pi) pi.GetPage(Of IPivotPage)).FirstOrDefault()
            If removedItem IsNot Nothing Then
                Await removedItem.OnPivotUnselectedAsync()
            End If

            If addedItem IsNot Nothing Then
                Await addedItem.OnPivotSelectedAsync()
            End If
        End Sub
    End Class
End Namespace
